pandas/tests/internals/test_external_block.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=W0102
+
+import numpy as np
+
+import pandas as pd
+from pandas.core.internals import Block, BlockManager
+
+
+class CustomBlock(Block):
+
+    def formatting_values(self):
+        return np.array(["Val: {}".format(i) for i in self.values])
+
+
+def test_custom_repr():
+    values = np.arange(3)
+
+    # series
+    block = CustomBlock(values, placement=slice(0, 3))
+    s = pd.Series(block, index=pd.RangeIndex(3), fastpath=True)
+    assert repr(s) == '0    Val: 0\n1    Val: 1\n2    Val: 2\ndtype: int64'
```

Comment:
so the windows test fail because this is int32 there. you have to use ``is_platform_windows()`` to make this test pass (e.g. use ``int32`` on windows, else use ``int64``). we do this somewhere else in the suite in test_format.py I think.